lib/manager/gomod/artifacts.ts

```diff
@@ -123,18 +175,17 @@ export async function updateArtifacts({
       extraEnv: {
         GOPATH: goPath,
         GOPROXY: process.env.GOPROXY,
-        GOPRIVATE: process.env.GOPRIVATE,
         GONOPROXY: process.env.GONOPROXY,
         GONOSUMDB: process.env.GONOSUMDB,
         GOFLAGS: useModcacherw(config.constraints?.go) ? '-modcacherw' : null,
         CGO_ENABLED: getAdminConfig().binarySource === 'docker' ? '0' : null,
+        ...getGoEnvironmentVariables(),
       },
       docker: {
         image: 'go',
         tagConstraint: config.constraints?.go,
         tagScheme: 'npm',
         volumes: [goPath],
-        preCommands: getPreCommands(),
```

Comment:
Why this?

Reply:
My idea was instead of running the preCommand to set global git configuration (`git config --global`), this configuration is injected via environment variables (`GIT_CONFIG_*`). The advantage of this is, that it works as well when running not in a sidecar docker container and is only applied to the command and not persisted. Allowing for more flexibility.